React/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -1,16 +1,45 @@
-export const actions = {
-  GET_LOGGED_USER: 'GET_LOGGED_USER',
-  SET_LOGGED_USER: 'SET_LOGGED_USER'
-};
+import { stringArrayToObject } from '../../utils/stringUtils';
+import { login as loginService } from '../../services/LoginService';
+
+export const actions = stringArrayToObject(
+  [
+    'LOGIN_REQUEST',
+    'LOGIN_SUCCESS',
+    'LOGIN_FAILURE'
+  ],
+  '@@LOGIN'
+);
 
 const actionCreators = {
-  getLoggedUser: () => ({
-    type: actions.GET_LOGGED_USER
-  }),
-  setLoggedUser: status => ({
-    type: actions.SET_LOGGED_USER,
-    logged: status
-  })
+  login: (user) => async dispatch => {
+    dispatch({ type: actions.LOGIN_REQUEST });
+    const response = await loginService({ email: user.email });
+    if (response.ok) {
+      if (response.data.length > 0) {
+        if (response.data[0].password === user.password) {
+          dispatch({
+            type: actions.LOGIN_SUCCESS,
+            payload: response.data
+          });
+        } else {
+          dispatch({
+            type: actions.LOGIN_FAILURE,
```

Comment:
i think the failure can be done in a ternary, because the only thing that will change is the payload..

Reply:
i tried that, with private action creators, but i think it looks not so good...

Reply:
yes, use privateActions. i'm sure you can have the payload with a ternary